orchestrator-salt/src/main/java/com/sequenceiq/cloudbreak/orchestrator/salt/SaltTelemetryOrchestrator.java

```diff
@@ -303,14 +315,55 @@ public void executeLoggingAgentDiagnostics(byte[] loggingAgentSaltState, List<Ga
             String loggingAgentZip = String.format("%s.zip", FLUENT_COMPONENT);
             uploadFileToTargetsWithContentAndPermission(sc, gatewayTargets, exitModel, loggingAgentSaltState,
                     REMOTE_TMP_FOLDER, loggingAgentZip, READ_WRITE_PERMISSION);
-            updateLoggingAgentSaltStateDefinition(sc, targets, loggingAgentZip);
+            updateSaltStateComponentDefinition(sc, targets, loggingAgentZip, FLUENT_COMPONENT);
             distributeAndExecuteLoggingAgentDoctor(allGateways, nodes, exitModel);
         } catch (Exception e) {
-            LOGGER.info("Error occurred during ", e);
+            LOGGER.info("Error occurred during logging agent diagnostics ", e);
             throw new CloudbreakOrchestratorFailedException(e.getMessage(), e);
         }
     }
 
+    @Override
+    public void updateMeteringSaltDefinition(byte[] meteringSaltState, List<GatewayConfig> allGateways, ExitCriteriaModel exitModel)
+            throws CloudbreakOrchestratorFailedException {
+        GatewayConfig primaryGateway = saltService.getPrimaryGatewayConfig(allGateways);
+        Set<String> gatewayTargets = getGatewayPrivateIps(allGateways);
+        Set<String> gatewayHostnames = getGatewayHostnames(allGateways);
+        try (SaltConnector sc = saltService.createSaltConnector(primaryGateway)) {
+            Target<String> targets = new HostList(gatewayHostnames);
+            uploadScripts(sc, gatewayTargets, exitModel, LOCAL_SALT_RESOURCES_LOCATION, SALT_STATE_UPDATER_SCRIPT);
+            String meteringZip = String.format("%s.zip", METERING_COMPONENT);
+            uploadFileToTargetsWithContentAndPermission(sc, gatewayTargets, exitModel, meteringSaltState,
+                    REMOTE_TMP_FOLDER, meteringZip, READ_WRITE_PERMISSION);
+            updateSaltStateComponentDefinition(sc, targets, meteringZip, METERING_COMPONENT);
+        } catch (Exception e) {
+            LOGGER.info("Error occurred during metering salt definition update.", e);
+            throw new CloudbreakOrchestratorFailedException(e.getMessage(), e);
+        }
+    }
+
+    @Override
+    public void upgradeMetering(List<GatewayConfig> allGateways, Set<Node> allNodes, ExitCriteriaModel exitModel, String upgradeFromDate, String customRpmUrl)
+            throws CloudbreakOrchestratorFailedException {
+        Map<String, Object> upgradeParameters = Map.of("metering", Map.of(
+                "upgradeDateFrom", upgradeFromDate,
+                "customRpmUrl", customRpmUrl,
+                "failHard", true));
+        GatewayConfig primaryGateway = saltService.getPrimaryGatewayConfig(allGateways);
+        Set<String> gatewayHostnames = getGatewayHostnames(allGateways);
+        try (SaltConnector sc = saltService.createSaltConnector(primaryGateway)) {
+            ParameterizedStateRunner stateRunner = new ParameterizedStateRunner(gatewayHostnames, allNodes,
```

Comment:
I'm not sure I understand why is this parameterized like this. The constructor looks like:
`public ParameterizedStateRunner(Set<String> targetHostnames, Set<Node> allNode, String state, Map<String, Object> parameters)`
and from how you create it, looks like this state would run only on the gateways instead of all the available instances
Could you clarify this?

Reply:
that was a bug